friends/src/components/FriendCard.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import EditForm from './EditForm';
+
+class FriendCard extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            shouldEdit: false,
+        }
+    }
+
+    beginEdit = () => {
+        if (!this.state.shouldEdit) {
+            this.setState({ shouldEdit: true });
+        } else {
+            this.setState({ shouldEdit: false });
+        }
```

Comment:
Look at the stuff that's the same in the logic of your `beginEdit` method here. How could we move the logic so that we don't repeat ourselves as much?
```javascript
// something like this:
beginEdit = () => {
  this.setState({ shoudlEdit: !this.state.shouldEdit })
};
```